proposals/0384-alpenglow-migration.md

```diff
@@ -0,0 +1,282 @@
+feature: a1penGLz8Vm2QHYB3JPefBiU4BY3Z6JkW2k3Scw5GWP
+---
+
+## Summary
+
+Migrate to Alpenglow from TowerBFT
+
+## Motivation
+
+Migrating from TowerBFT to Alpenglow consensus requires a safe handoff
+mechanism that doesn't rollback TowerBFT confirmed user transactions.
+
+## Dependencies
+
+This proposal depends on the following accepted proposal:
+
+- **[SIMD-0326]: Alpenglow**
+
+    Requires Alpenglow to be implemented in order to migrate
+
+- **[SIMD-0307]: Add Block Footer**
+
+    Specifies `BlockMarker`, an means of disseminating metadata in a block
+
+
+## New Terminology
+
+Migration boundary slot:
+
+- Slot at which Alpenglow migration begins
+- After slot this boundary slot:
+   - Turn off packing anything other than simple vote transactions in blocks.
+   Any transactions not
+     belonging to the vote program will cause a block to be marked dead during
+     replay by all correct validators.
+   - Core code will stop notifying RPC services of any optimistic confirmation/
+   commitment updates.
+   - TowerBFT will stop rooting blocks to prevent losing optimistically
+     confirmed blocks that could qualify as the Alpenglow genesis blocks.
+
+Alpenglow genesis block:
+
+- The last TowerBFT block before the "migration boundary slot" which is the
+parent of the first Alpenglow block. This is picked via the process described in
+the "Detailed Design" section below
+
+Strong optimistic confirmation:
+
+- A block `B` is strong OC if there exists a block `T` such that `B` is the parent
+  of `T` and `slot(B) + 1 = slot(T)` and `T` contains vote transactions for `B` from
+  at least `82%` of stake.
+
+## Detailed Design
+
+### Migration Handoff
+
+1. Pick a "migration boundary slot" (defined above) `S` as follows. Let `X` be
+the rooted slot in
+   which the feature flag is activated. Let the migration slot S be `X + 5000`,
+   as to avoid the beginning of an epoch.
+
+2. After the migration boundary slot `S`, wait for some block `B > S` to reach
+   strong optimistic confirmation.
+
+3. Find the latest ancestor block `G` of `B` from before the migration boundary
+   slot `S`. This is the Alpenglow genesis block. Cast a BLS vote, the "genesis
+   vote", for `G` via all to all. Note validators should have filled out
+   their BLS keys prior to the feature flag activation, and will sign this
+   genesis vote with this BLS key.
+
+4. If we observe `>=82%` genesis votes for the ancestor block `G`, this
+   consitutes the `genesis certificate`, and `G` is the genesis block for
+   Alpenglow. Validators will periodically refresh genesis votes every
+   `GENESIS_VOTE_REFRESH` = 400ms (i.e. once a slot) until this
+   `genesis certificate` is observed. During this period they perform regular
+   TowerBFT consensus for all blocks.
+
+5. Anytime a correct validator receives a `genesis certificate` for a slot `G`
+   (either constructed themselves, received through replaying a block, or received
+   from all-to-all broadcast), they:
```

Comment:
You didn't say but I suppose the first thing they will do (if they received it rather than constructed it) would be verify the certificate is valid. (And do we need to list validity checks?)

Reply:
updated!